packages/teams-js/src/public/appWindow.ts

```diff
@@ -77,6 +78,7 @@ export class ParentAppWindow implements IAppWindow {
    * @param listener - The listener that will be called
    */
   public addEventListener(type: string, listener: (message: any) => void): void {
+    ensureInitialized();
```

Comment:
Without passing anything as `expectedFrameContexts`, we don't validate any of them and instead just make sure that initialize has been called, the same way we do in `postMessage`. I think that is the right behavior here, unless I'm missing something.

Reply:
I don't think this function makes any sense to call except from within a dialog/task, so I guess I was wondering should we verify that that is the frame context it's being used inside of? I think the function won't do anything in other contexts.

Reply:
I see what you mean. Yes, in this case we should pass `FrameContexts.task` to it.

Reply:
All this will become deprecated with the work Lakhveer is doing on moving these to `dialog` capability directly, right? In that case, do we really want/need to update it, or should we only make sure the new APIs have the right checks in place?

Reply:
Yes, it is deprecated, but the card based dialogs will still be using tasks.ts instead of dialog.ts.